tools/SourceKit/include/SourceKit/Core/LangSupport.h

```diff
@@ -20,8 +20,10 @@
 #include "llvm/ADT/Optional.h"
 #include "llvm/ADT/SmallString.h"
 #include "swift/AST/Type.h"
+#include "swift/Syntax/SyntaxNodes.h"
```

Comment:
Could you forward declare `swift::syntax::SourceFileSyntax` instead of `#include`? Modifying `SyntaxNodes.h`(or `gyb_syntax_support` files) may cause almost entire `SourceKit` rebuild.

Reply:
Would maybe be a good idea.

We currently already need to rebuild everything if we change the `gyb_syntax_support` because `AST/Module.h` includes `Syntax/SyntaxParsingCache.h` and that includes `SyntaxNodes.h`. I should probably break that dependency as well.

Reply:
Even if it make little difference for now, I generally prefer forward declare in `.h`, include in `.cpp` where possible.

Reply:
OK. I'll leave it at the forward declaration for now then.